hil/src/commands/ota.rs

```diff
@@ -0,0 +1,634 @@
+use std::path::PathBuf;
+use std::time::{Duration, Instant};
+
+use crate::serial::{spawn_serial_reader_task, LOGIN_PROMPT_PATTERN};
+use crate::ssh_wrapper::SshWrapper;
+use clap::Parser;
+use color_eyre::{
+    eyre::{bail, eyre, WrapErr},
+    Result,
+};
+use futures::StreamExt;
+use regex::Regex;
+use serde_json::Value;
+use tokio::sync::broadcast;
+use tokio_serial::SerialPortBuilderExt;
+use tokio_stream::wrappers::BroadcastStream;
+use tracing::{debug, error, info, instrument, warn};
+
+/// Over-The-Air update command for the Orb
+#[derive(Debug, Parser)]
+pub struct Ota {
+    /// Target version to update to
+    #[arg(long)]
+    version: String,
+
+    /// Hostname of the Orb device
+    #[arg(long)]
+    host: String,
+
+    /// Username
+    #[arg(long, default_value = "worldcoin")]
+    username: String,
+
+    /// Password
+    #[arg(long)]
+    password: String,
+
+    /// SSH port for the Orb device
+    #[arg(long, default_value = "22")]
+    port: u16,
+
+    /// Platform type (diamond or pearl)
+    #[arg(long, value_enum)]
+    platform: Platform,
+
+    /// Timeout for the entire OTA process in seconds
+    #[arg(long, default_value = "7200")] // 2 hours by default
+    timeout_secs: u64,
+
+    /// Path to save journalctl logs from worldcoin-update-agent.service
+    #[arg(long)]
+    log_file: PathBuf,
+
+    /// Maximum reconnection attempts after reboot
+    #[arg(long, default_value = "50")]
+    max_reconnect_attempts: u32,
+
+    /// Sleep time between reconnection attempts in seconds
+    #[arg(long, default_value = "5")]
+    reconnect_sleep_secs: u64,
+
+    /// Serial port path for boot log capture
+    #[arg(long, default_value = "/dev/ttyUSB1")]
+    serial_path: PathBuf,
+}
+
+#[derive(Debug, Clone, clap::ValueEnum)]
+enum Platform {
+    Diamond,
+    Pearl,
+}
+
+impl Ota {
+    #[instrument(skip(self))]
```

Comment:
```suggestion
```
I wouldn't skip instrumenting the arguments here. Its good to know what the command was run with when seeing the traces.